app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt

```diff
@@ -293,58 +252,61 @@ class ThreadFragment : Fragment() {
     }
 
     private fun setupAdapter(result: OpenThreadResult) = with(binding.messagesList) {
-
         addItemDecoration(DividerItemDecorator(InsetDrawable(dividerDrawable(context), 0)))
         recycledViewPool.setMaxRecycledViews(0, 0)
+        threadAdapter.stateRestorationPolicy = StateRestorationPolicy.PREVENT_WHEN_EMPTY
+        updateAdapter(result)
+    }
 
-        threadAdapter.apply {
-            isExpandedMap = result.isExpandedMap
-            initialSetOfExpandedMessagesUids = result.initialSetOfExpandedMessagesUids
-            isThemeTheSameMap = result.isThemeTheSameMap
-
-            stateRestorationPolicy = StateRestorationPolicy.PREVENT_WHEN_EMPTY
-            contacts = mainViewModel.mergedContactsLive.value ?: emptyMap()
-        }
+    private fun setupDialogs() {
+        bindAlertToViewLifecycle(descriptionDialog)
+        linkContextualMenuAlertDialog.initValues(mainViewModel.snackBarManager)
+        emailContextualMenuAlertDialog.initValues(mainViewModel.snackBarManager)
+        phoneContextualMenuAlertDialog.initValues(mainViewModel.snackBarManager)
```

Comment:
```suggestion
        mainViewModel.snackBarManager.apply {
              linkContextualMenuAlertDialog.initValues(this)
              emailContextualMenuAlertDialog.initValues(this)
              phoneContextualMenuAlertDialog.initValues(this)
        }
```

Reply:
`apply` is used to modify the object.
Here, it's not modified, it's only used.
It should be `with` instead.

Nevertheless, we tend to avoid using too much `this`, it can be confusing.
I propose this, if it's ok for you:
```kt
private fun setupDialogs() = with(mainViewModel) {
    bindAlertToViewLifecycle(descriptionDialog)
    linkContextualMenuAlertDialog.initValues(snackBarManager)
    emailContextualMenuAlertDialog.initValues(snackBarManager)
    phoneContextualMenuAlertDialog.initValues(snackBarManager)
}
```